### DISCUSSION TITLE
Socket io (Error: server error) while in production - Using ECS Fargate with Elasticache Redis

### DISCUSSION THREAD

**Discussion by ankitdsgn** [socketio/socket.io#5137] [Q&A] [answered]
I have been working on a chat application and everything seems to work in localhost but when I am in production I am having difficulty connecting my socket io client with my socket server.

This is the error I am getting while making the connection

```
Error: server error
    at zq.onPacket (socket.js:335:33)
    at uq.emit (index.mjs:136:20)
    at polling.onPacket (transport.js:98:15)
    at polling.js:116:18
    at Array.forEach (<anonymous>)
    at polling.onData (polling.js:119:53)
    at uq.emit (index.mjs:136:20)
    at Nq.onLoad (polling.js:352:18)
    at n.onreadystatechange (polling.js:288:26)
    at XMLHttpRequest.r (helpers.ts:98:17)
```
But in my network section am getting ```200 ok status```

![Screenshot 2024-07-17 170123](https://github.com/user-attachments/assets/12b2c16a-9abc-4750-86d5-bdfc8834d86e)

This is my client side code and my REACT_APP_SOCKET_URL is ```https://test.mycompany.com```

   ```
 socket = io(REACT_APP_SOCKET_URL, {
      rejectUnauthorized: false,
      query: {
        token: token,
      },
    });
  }

  useEffect(() => {
    socket.on("connect_error", (err) => {
      console.log(err);
    });

    socket.emit("member_online", {
      _id,
      teamId: sessionStorage.getItem("teamId"),
    });
```
My server side code
```
io = new Server(mainServer, {
      cors: {
        origin: "*", // Replace with your client's URL
        methods: ["GET", "POST"],
      },
    });
    io.adapter(createShardedAdapter(pubClient, subClient));
  } catch (error) {
    console.error("Error creating Redis adapter: ", error);
  }

  io.on("connection", (socket) => {
    socket.on("connect_error", (err) => {
      console.log(`connect_error due to ${err.message}`);
    });
    socket.on("member_online", async (data) => {
      console.log("member_online", data);
```

What have I tried?

I have tried many things actually. I thought there was something with CORS so enabled everything with ```*```, I also tried adding the port number to my url something like this: ```https://test.mycompany.com:5000``` still it did not work.

I also disabled the ```ssl``` with ```rejectUnauthorized: false```

Really could use some help here. Really getting frustrated.

**Comment by darrachequesne** [comment]
Hi!

That's weird indeed. The "server error" is created here: https://github.com/socketio/socket.io/blob/b9b16132c2042494f946413ecc687edfa10e40cb/packages/engine.io-client/lib/socket.ts#L589-L592

Which means the client has received a malformed packet (the "error" packet type is not used during the communication, only when parsing). Could you please share the body of the HTTP request?

**Comment by ankitdsgn** [comment, marked as answer]
Very silly mistake on my behalf. I forgot that I have nginx reverse proxy in place. So I had to add a section for socket io in my nginx default.conf file.

```
location /socket.io/ {
        proxy_pass http://127.0.0.1:5000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_set_header Host $host;
        proxy_hide_header 'Access-Control-Allow-Origin';
        }
```
And it worked like a charm.
